sdk/storage/storage-blob/src/policies/StorageCorrectConentLengthPolicy.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+  PipelinePolicy,
+} from "@azure/core-rest-pipeline";
+import { HeaderConstants } from "../utils/constants";
+import { isNode } from "@azure-tools/test-utils";
+
+/**
+ * The programmatic identifier of the storageCorrectContentLengthPolicy.
+ */
+export const storageCorrectContentLengthPolicyName = "StorageCorrectContentLengthPolicy";
+
+
+/**
+ * storageCorrectConentLengthPolicy to correctly set Content-Length header with request body length.
+ */
+export function storageCorrectContentLengthPolicy(): PipelinePolicy {
+  function correctContentLength(request: PipelineRequest): void {
+    if (isNode) {
```

Comment:
Can we move this check to `Pipeline.ts` and conditionally add the policy? Otherwise referencing `Buffer` might cause trouble.
Alternatively, we could make a .browser version of this file that is a no-op